demos/glacial_isostatic_adjustment/base_case/base_case.py

```diff
@@ -14,191 +14,237 @@
+#
+# Once these spaces have been chosen, we can integrate the divergence of the incremental Lagrangian stress term by parts within each element, $K_n$, to introduce weak boundary conditions and to move derivatives from the trial function to the test function according to
+#
 # \begin{equation}
-#     \nabla \cdot \boldsymbol{\sigma}_{L1} - \nabla (\rho_0 g u_r)   - \rho_1 \textbf{g}  = 0,
+#     S  = \sum_n \left(\int_{K_n} \nabla \boldsymbol{\phi} : \boldsymbol{\sigma}_1^L  \, dx - \int_{\partial K_n}  \boldsymbol{\phi} \cdot \left( \hat{\boldsymbol{n}} \cdot \boldsymbol{\sigma}_1^L  \right) \, ds \right) ,
 # \end{equation}
+#
+# where $\hat{\boldsymbol{n}}$ is the outward pointing normal vector to an element edge, represented by $\partial K_n$. Given that test functions are continuous across cell edges and we assume that there is no jump in stress across material discontinuities this simplifies to
+#
 # \begin{equation}
-#     \nabla \cdot \textbf{v} = 0,
+#     S  = \int_{\Omega}  \nabla \boldsymbol{\phi} : \left( \kappa \nabla \cdot \boldsymbol{u}\, \boldsymbol{I}  + 2 \mu_0 \boldsymbol{d}(\boldsymbol{u})   - 2 \sum_i \mu_i \boldsymbol{m}_i(\boldsymbol{u})   \right) \, dx
+#      + \int_{\partial \Omega_\textrm{top}}  \boldsymbol{\phi} \cdot \left( B_{\mu} \, \rho_\mathrm{load} \, g \, h_\mathrm{load}  \hat{\boldsymbol{n}} \right) \, ds ,
 # \end{equation}
+#
+# where we have substituted the constitutive relation and the boundary condition at the top of the domain, $\partial \Omega_\textrm{top}$. For bottom (and side) boundaries where we specify no normal displacement, we utilise two different strategies depending on domain geometry. In Cartesian domains, we apply the boundary condition strongly by modifying the discrete test and trial spaces, such that the surface integral vanishes. For spherical domains, we implement that condition weakly using the Symmetric Interior Penalty Galerkin method.
+#
+# The same approach is taken for advection of the hydrostatic pre-stress term, initially integrating by parts to give
+#
 # \begin{equation}
-#     \rho_1 = - \textbf{u} \cdot \nabla \rho_0,
+#     H  = \sum_i \left(\int_{K_i} -\nabla \cdot \boldsymbol{\phi}~   B_{\mu} \rho_0 g u_k  \, dx  + \int_{\partial K_i} \boldsymbol{\phi} \cdot \hat{\boldsymbol{n}}  B_{\mu} \rho_0 g u_k \, ds  \right).
 # \end{equation}
+#
+# Since we allow density and gravity discontinuities through the $DG0$ discretisation, we need to also account for the contribution of surface integrals along interior facets, denoted by $\Gamma$, using
+#
+# \begin{equation}
+#     H  = -\int_{\Omega} \nabla \cdot \boldsymbol{\phi} B_{\mu} \rho_0 g u_k \, dx  + \int_{\Gamma} \boldsymbol{\phi} \cdot B_{\mu} u_k [[\rho_0 g \hat{\boldsymbol{n}}]]  \, ds  + \int_{\partial \Omega_\textrm{top}} \boldsymbol{\phi} \cdot \hat{\boldsymbol{n}} B_{\mu} \rho_0 g u_k \, ds  ,
+# \end{equation}
+#
+# where the jump term is given by
```

Comment:
I suspect its worth specifying somewhere above what you really mean by jump term here. I'm kinda clear but also unclear. Remember the tutorial readers will be from a range of backgrounds and they'll need to understand the significance of these jump terms.